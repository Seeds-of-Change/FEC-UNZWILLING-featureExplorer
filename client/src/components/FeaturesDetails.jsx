import React from 'react';
import PropTypes from 'prop-types';

class FeaturesDetails extends React.Component {
  constructor(props) {
    super(props);
    const { features } = this.props;
    this.state = {
      focusedFeature: features[0],
    };
  }

  render() {
    const { header, features } = this.props;
    return (
      <div className="FeaturesDetails">
        <h3 className="FeaturesDetails-header">{header}</h3>
        {features.map((feature, index) => (
          <div key={`${feature.posX}${feature.posY}`}>
            <h4 className="FeaturesDetails-feature-heading">
              {feature.heading}
            </h4>
            <div className="FeaturesDetails-feature-description">
              {feature.description}
            </div>
          </div>
        ))}
      </div>
    );
  }
}

FeaturesDetails.defaultProps = {
  header: null,
  features: null,
};

FeaturesDetails.propTypes = {
  features: PropTypes.arrayOf(
    PropTypes.shape({
      heading: PropTypes.string,
      description: PropTypes.string,
    }),
  ),
  header: PropTypes.string,
};

export default FeaturesDetails;
