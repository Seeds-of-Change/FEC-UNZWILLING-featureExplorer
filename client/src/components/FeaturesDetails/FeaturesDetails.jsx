import React from 'react';
import PropTypes from 'prop-types';
import style from './styles.css';

export default class FeaturesDetails extends React.Component {
  constructor(props) {
    super(props);
    const { features } = this.props;
    this.state = {
      activeIndex: "0",
    };
    this.handleFeatureChange = this.handleFeatureChange.bind(this);
  }

  handleFeatureChange(e) {
    // const { features } = this.props;
    const newActiveIndex = e.target.parentNode.attributes.index.value;

    this.setState({ activeIndex: newActiveIndex });
  }

  render() {
    const { header, features } = this.props;
    const { activeIndex } = this.state;
    return (
      <div className={style.wrapper}>
        <h3 className={style.header}>{header}</h3>
        <ul className={style.features}>
          {features.map((feature, index) => (
            <li className={this.state.activeIndex == index ? `${style.feature} ${style.active}` : `${style.feature}`} key={`${feature.posX}${feature.posY}`} index={index}>
              <h4 className={style.feature_heading} onClick={this.handleFeatureChange}>
                {feature.heading}
              </h4>
              <div className={style.feature_description}>
                {feature.description}
              </div>
            </li>
          ))}

        </ul>
      </div>
    );
  }
}

FeaturesDetails.defaultProps = {
  header: null,
  features: null,
};

FeaturesDetails.propTypes = {
  features: PropTypes.arrayOf(
    PropTypes.shape({
      heading: PropTypes.string,
      description: PropTypes.string,
    }),
  ),
  header: PropTypes.string,
};

// export default FeaturesDetails;
