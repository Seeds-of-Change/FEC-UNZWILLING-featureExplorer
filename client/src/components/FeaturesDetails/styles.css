.wrapper {
  width: 45%;
  display:inline-block;
  vertical-align: top;
}

.header{
  color: rgb(39, 34, 29);
  display: block;
  font-size: 36px;
  font-weight: 700;
  line-height: 36px;
  margin-bottom: 60px;
  text-transform: uppercase;
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
}

.features{
  list-style: none;
  padding: 0;
}

.feature{
  margin-bottom: 50px;
}

.feature_heading, .feature_heading:focus{
  color: rgb(153, 150, 147);
  font-size: 1.5rem;
  line-height: 36px;
  cursor: pointer;
  border: none;
  background: none;
  display: block;
  margin: 0;
  padding: 0;
  outline: none;
  text-align: left;
}

.feature_description{
  font-size: 1rem;
  line-height: 24px;
  margin-top: 20px;
  display: none;
}

.active > .feature_heading{
  color: black;
}
.active > .feature_description{
  display: block;
}