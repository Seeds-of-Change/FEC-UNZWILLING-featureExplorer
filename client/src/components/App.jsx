import React from 'react';
import PropTypes from 'prop-types';
import FeaturesDetails from './FeaturesDetails';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      features: [{ value: '1' }, { value: '2' }, { value: '3' }],
    };
  }

  render() {
    const { features } = this.state;
    return (
      <div className="App">
        <FeaturesDetails features={features} />

      </div>
    );
  }
}

export const Counter = ({ counter }) => (
  <div>
    <p>{ counter }</p>
  </div>
);

App.propTypes = {

};

export default App;
