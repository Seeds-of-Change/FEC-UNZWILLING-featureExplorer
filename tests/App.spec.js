import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import App from '../client/src/components/App';
import FeaturesDetails from '../client/src/components/FeaturesDetails';

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders the Features Details Component', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find(FeaturesDetails).length).toEqual(1);
  });
});

describe('My Test Suite', () => {
  test('My Test Case', () => {
    expect(true).toEqual(true);
  });
});