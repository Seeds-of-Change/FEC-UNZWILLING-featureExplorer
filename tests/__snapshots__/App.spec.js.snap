// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="App"
>
  <div
    className="FeaturesDetails"
  >
    <h3
      className="FeaturesDetails-header"
    >
      
    </h3>
  </div>
</div>
`;
